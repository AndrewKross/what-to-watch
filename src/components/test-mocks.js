export const PromoFilmData = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`
};

const GENRE = `drama`;
const RELEASE = `2020`;
const RATING = 9;
const VOTES = 228;
const DIRECTOR = `Otto Bathurst`;
const PREVIEW = `https://cdn.videvo.net/videvo_files/video/free/2019-09/small_watermarked/190828_07_MarinaBayatNightDrone_UHD_02_preview.webm`;
const RUN_TIME = 140;

const DESCRRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's
friend and protege. Gustave prides himself on providing first-class service to the hotel's guests,
including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's
lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief
suspect in her murder.`;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`
];

const COVERS = [
  `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `img/bohemian-rhapsody.jpg`,
  `img/macbeth.jpg`,
  `img/aviator.jpg`,
  `img/we-need-to-talk-about-kevin.jpg`,
  `img/what-we-do-in-the-shadows.jpg`,
  `img/revenant.jpg`,
  `img/johnny-english.jpg`
];

const ACTORS = [
  `Cillian Murphy`,
  `Sam Neill`,
  `Helen McCrory`,
  `Paul Anderson`,
  `Annabelle Wallis`,
  `Iddo Goldberg`
];

const review = {
  id: `0`,
  text: `I didn't find it amusing, and while I can appreciate the creativity,
  it's an hour and 40 minutes I wish I could take back.`,
  rating: 8,
  userName: `Kate Muir`,
  date: new Date()
};

export const films = TITLES.map((title, index) => {
  return ({
    id: index.toString(),
    title,
    cover: COVERS[index],
    poster: COVERS[index],
    preview: PREVIEW,
    genre: GENRE,
    release: RELEASE,
    rating: RATING,
    votes: VOTES,
    description: DESCRRIPTION,
    director: DIRECTOR,
    actors: ACTORS,
    runTime: RUN_TIME,
    reviews: [review]
  });
});

export const TABS_DATA = [
  `overview`,
  `details`,
  `reviews`
];
