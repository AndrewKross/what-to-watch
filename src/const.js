export const NUMBER_OF_SIMILAR_FILMS = 4;
export const PREVIEW_DELAY = 1000;
export const REVIEWS_IN_COLUMN_COUNT = 3;
export const FILMS_ON_START_SCREEN = 8;
export const FILMS_ON_SHOW_MORE_BUTTON = 20;
export const REVIEW_DATE_HUMAN_FORMAT = `MMMM D, YYYY`;
export const REVIEW_DATE_SERVICE_FORMAT = `YYYY-MM-DD`;

export const RatingRange = {
  MIN: 0,
  MAX_BAD: 3,
  MAX_NORMAL: 5,
  MAX_GOOD: 8,
  MAX: 10,
};

export const RatingGrade = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const AppRoute = {
  MAIN: `/`,
  FILM: `/film/`,
  PLAYER: `/player/`,
};

export const GENRES = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`,
];

export const GenresForFilms = {
  [`Comedies`]: `comedy`,
  [`Crime`]: `crime`,
  [`Documentary`]: `documentary`,
  [`Dramas`]: `drama`,
  [`Horror`]: `horror`,
  [`Kids & Family`]: `kids & family`,
  [`Romance`]: `romance`,
  [`Sci-Fi`]: `sci-fi`,
  [`Thrillers`]: `thriller`,
};

export const TabsNames = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export const TabsData = [{
  name: TabsNames.OVERVIEW,
  content: `Overview`,
},
{
  name: TabsNames.DETAILS,
  content: `Details`,
},
{
  name: TabsNames.REVIEWS,
  content: `Reviews`,
},
];
