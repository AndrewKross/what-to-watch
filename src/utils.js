export const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min;
};

export const getRatingLevel = (rating) => {
  rating = parseInt(rating, 10);
  if (rating >= 0 && rating < 3) {
    return `Bad`;
  }
  if (rating < 5) {
    return `Normal`;
  }
  if (rating < 8) {
    return `Good`;
  }
  if (rating < 10) {
    return `Very good`;
  }
  if (rating === 10) {
    return `Awesome`;
  }
  return null;
};
