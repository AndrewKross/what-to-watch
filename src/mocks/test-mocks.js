import { NameSpace } from '../reducer/reducer';
import { DEFAULT_GENRE, LoadingStatus } from '../const';

const RELEASE = 2020;
const RATING = 9;
const RATINGS_COUNT = 228;
const DIRECTOR = `Otto Bathurst`;
const PREVIEW = `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`;
const RUN_TIME = 140;
const COMMENTS_COUNT = 6;

const DESCRIPTION = `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's
friend and protege. Gustave prides himself on providing first-class service to the hotel's guests,
including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's
lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief
suspect in her murder.`;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
];

const COVERS = [
  `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `img/bohemian-rhapsody.jpg`,
  `img/macbeth.jpg`,
  `img/aviator.jpg`,
  `img/we-need-to-talk-about-kevin.jpg`,
  `img/what-we-do-in-the-shadows.jpg`,
  `img/revenant.jpg`,
  `img/johnny-english.jpg`,
];

const ACTORS = [
  `Cillian Murphy`,
  `Sam Neill`,
  `Helen McCrory`,
  `Paul Anderson`,
  `Annabelle Wallis`,
  `Iddo Goldberg`,
];

export const GENRES = [
  `all genres`,
  `comedy`,
  `drama`,
  `melodrama`,
  `horror`,
  `crime`,
  `documentary`,
  `romance`,
  `thriller`,
];

export const TABS_DATA = [
  `overview`,
  `details`,
  `reviews`,
];

const comment = {
  id: 0,
  text: `I didn't find it amusing, and while I can appreciate the creativity,
  it's an hour and 40 minutes I wish I could take back.`,
  rating: 8,
  userName: `Kate Muir`,
  date: new Date(`December 25, 2019 01:00:00`),
};

export const comments = Array(COMMENTS_COUNT)
  .fill(``)
  .map(() => comment);

export const films = TITLES.map((title, index) => {
  return ({
    id: index,
    title,
    cover: COVERS[index],
    posterImage: COVERS[index],
    previewImage: COVERS[index],
    backgroundImage: COVERS[index],
    previewVideo: PREVIEW,
    videoMain: PREVIEW,
    genre: GENRES[index + 1] || GENRES[1],
    released: RELEASE,
    rating: RATING,
    ratingsCount: RATINGS_COUNT,
    description: DESCRIPTION,
    director: DIRECTOR,
    starring: ACTORS,
    runTime: RUN_TIME,
    isFavorite: false,
    backgroundColor: `black`,
  });
});

export const userData = {
  id: 1,
  email: `Andrew@gmail.com`,
  name: `Andrew`,
  avatar: `img/1.png`,
};

export const loginData = {
  email: `Andrew@gmail.com`,
  password: `1234`,
};

export const reviewToPost = {
  rating: 8,
  comment: `Great!`,
};

export const defaultMockStore = {
  [NameSpace.DATA]: {
    films,
    comments,
    isFilmsLoaded: true,
    isCommentsLoaded: true,
  },
  [NameSpace.APP_STATE]: {
    filmsOnScreen: 8,
    currentGenre: DEFAULT_GENRE,
  },
  [NameSpace.USER]: {
    isAuthorized: true,
    userInfo: userData,
    reviewLoadingStatus: LoadingStatus.PENDING,
    authorizationLoadingStatus: LoadingStatus.PENDING,
  },
};
