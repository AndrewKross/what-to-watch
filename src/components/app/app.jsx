import React from 'react';
import PropTypes from 'prop-types';
import {
  HashRouter, Link, Route, Router, Switch,
} from 'react-router-dom';
import { connect } from 'react-redux';
import Loader from 'react-loader-spinner';
import { AppRoute, HttpStatus, LoaderData } from '../../const';
import MainPage from '../main-page/main-page.jsx';
import FilmPage from '../film-page/film-page.jsx';
import MainPlayer from '../main-player/main-player.jsx';
import {
  ActionCreator as DataActionCreator,
  Operation as DataOperation,
} from '../../reducer/data/data';
import {
  getComments, getFilms, getFilmsLoadingStatus, getPromoFilm, getPromoFilmLoadState,
  getRequestStatus,
} from '../../reducer/data/selectors';
import { getFilmsOnScreen, getFilteredFilms } from '../../reducer/app-state/selectors';
import SignIn from '../sign-in/sign-in.jsx';
import { getFilmFromRoute } from '../../utils/films';
import AddReview from '../add-review/add-review.jsx';
import { getAuthorizationStatus } from '../../reducer/user/selectors';
import history from '../../history';
import UserPage from '../user-page/user-page.jsx';
import Error from '../error/error.jsx';

const App = ({
  filteredFilms, filmsOnScreen, comments, loadComments,
  isAuthorized, isFilmsLoaded, films, requestStatus, promoFilm, isPromoFilmLoaded,
}) => {
  if (!isFilmsLoaded && !isPromoFilmLoaded) {
    return (
      <div style={LoaderData.STYLE}>
        <Loader
          type={LoaderData.TYPE}
          color={LoaderData.COLOR}
          width={LoaderData.HEIGHT}
          height={LoaderData.WIDTH}
        />
      </div>
    );
  }

  if (requestStatus !== HttpStatus.SUCCESS) {
    return (
      <Error requestStatus={requestStatus}/>
    );
  }

  return (
    <HashRouter>
      <Router history={history}>
        <Switch>
          <Route exact path={AppRoute.MAIN}
                 render={() => {
                   return (<MainPage
                     filteredFilms={filteredFilms}
                     filmsOnScreen={filmsOnScreen}
                     promoFilm={promoFilm}
                   />);
                 }}
          />
          <Route exact path={`${AppRoute.FILM}:id`}
                 render={({ match }) => {
                   return (<FilmPage
                     selectedFilm={getFilmFromRoute(films, match)}
                     comments={comments}
                     loadComments={loadComments}
                     films={filteredFilms}
                     filmsOnScreen={filmsOnScreen}
                     isAuthorized={isAuthorized}
                   />);
                 }}
          />
          <Route exact path={`${AppRoute.FILM}:id/review`}
                 render={({ match }) => (isAuthorized
                   ? <AddReview film={getFilmFromRoute(filteredFilms, match)}/>
                   : <SignIn/>)
                 }
          />
          <Route exact path={`${AppRoute.PLAYER}:id`}
                 render={({ match }) => <MainPlayer film={getFilmFromRoute(filteredFilms, match)}/>}
          />
          <Route exact path={AppRoute.LOGIN} component={SignIn}/>
          <Route exact path={AppRoute.MY_LIST}
                 render={() => (isAuthorized
                   ? <UserPage/>
                   : <SignIn/>)}/>
          <Route
            render={() => (
              <React.Fragment>
                <h1>
                  Error: 404. Page not found.
                </h1>
                <Link to={AppRoute.MAIN}>Go to main page</Link>
              </React.Fragment>
            )}
          />
        </Switch>
      </Router>
    </HashRouter>
  );
};

const mapStateToProps = (state) => ({
  films: getFilms(state),
  filteredFilms: getFilteredFilms(state),
  filmsOnScreen: getFilmsOnScreen(state),
  promoFilm: getPromoFilm(state),
  comments: getComments(state),
  isAuthorized: getAuthorizationStatus(state),
  isFilmsLoaded: getFilmsLoadingStatus(state),
  requestStatus: getRequestStatus(state),
  isPromoFilmLoaded: getPromoFilmLoadState(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadComments(filmId) {
    dispatch(DataActionCreator.changeCommentsLoadState(false));
    dispatch(DataOperation.loadComments(filmId));
  },
});

App.propTypes = {
  films: PropTypes.array.isRequired,
  filmsOnScreen: PropTypes.number.isRequired,
  filteredFilms: PropTypes.array.isRequired,
  comments: PropTypes.array.isRequired,
  loadComments: PropTypes.func.isRequired,
  isAuthorized: PropTypes.bool.isRequired,
  isFilmsLoaded: PropTypes.bool.isRequired,
  requestStatus: PropTypes.number.isRequired,
  promoFilm: PropTypes.object.isRequired,
  isPromoFilmLoaded: PropTypes.bool.isRequired,
};

export { App as AppComponent };
export default connect(mapStateToProps, mapDispatchToProps)(App);
