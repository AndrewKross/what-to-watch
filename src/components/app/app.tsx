import * as React from 'react';
import Loader from 'react-loader-spinner';
import { connect } from 'react-redux';
import { Link, Route, Router, Switch } from 'react-router-dom';
import { AppRoute, HttpStatus, LoaderData } from '../../const';
import history from '../../history';
import { getFilmsOnScreen, getFilteredFilms } from '../../reducer/app-state/selectors';
import {
  ActionCreator as DataActionCreator,
  Operation as DataOperation,
} from '../../reducer/data/data';
import {
  getComments,
  getFilms,
  getFilmsLoadingStatus,
  getPromoFilm,
  getPromoFilmLoadState,
  getRequestStatus,
} from '../../reducer/data/selectors';
import { getAuthorizationStatus } from '../../reducer/user/selectors';
import { Comment, Film } from '../../types';
import { getFilmFromRoute } from '../../utils/films';
import AddReview from '../add-review/add-review';
import FilmPage from '../film-page/film-page';
import MainPage from '../main-page/main-page';
import MainPlayer from '../main-player/main-player';
import SignIn from '../sign-in/sign-in';
import UserPage from '../user-page/user-page';

interface Props {
  films: Film[],
  filmsOnScreen: number,
  filteredFilms: Film[],
  comments: Comment[]
  loadComments: () => void,
  isAuthorized: boolean,
  isFilmsLoaded: boolean,
  requestStatus: number,
  promoFilm: Film,
  isPromoFilmLoaded: boolean,
}

const App: React.FunctionComponent<Props> = ({
  filteredFilms, filmsOnScreen, comments, loadComments, isAuthorized, isFilmsLoaded,
  films, requestStatus, promoFilm, isPromoFilmLoaded,
}: Props) => {
  if (!isFilmsLoaded || !isPromoFilmLoaded) {
    return (
      <div style={LoaderData.STYLE}>
        <Loader
          type={LoaderData.TYPE}
          color={LoaderData.COLOR}
          width={LoaderData.HEIGHT}
          height={LoaderData.WIDTH}
        />
      </div>
    );
  }

  if (requestStatus !== HttpStatus.SUCCESS) {
    return (
      <h1>Error: {requestStatus}</h1>
    );
  }

  return (
    <Router history={history}>
      <Switch>
        <Route exact path={AppRoute.MAIN}
               render={() => {
                 return (<MainPage
                   filteredFilms={filteredFilms}
                   filmsOnScreen={filmsOnScreen}
                   promoFilm={promoFilm}
                 />);
               }}
        />
        <Route exact path={`${AppRoute.FILM}:id`}
               render={({ match }) => {
                 return (<FilmPage
                   selectedFilm={getFilmFromRoute(films, match)}
                   comments={comments}
                   loadComments={loadComments}
                   films={filteredFilms}
                   isAuthorized={isAuthorized}
                 />);
               }}
        />
        <Route exact path={`${AppRoute.FILM}:id/review`}
               render={({ match }) => (isAuthorized
                 ? <AddReview film={getFilmFromRoute(filteredFilms, match)}/>
                 : <SignIn/>)
               }
        />
        <Route exact path={`${AppRoute.PLAYER}:id`}
               render={({ match }) => <MainPlayer
                 film={getFilmFromRoute(filteredFilms, match)}/>}
        />
        <Route exact path={AppRoute.LOGIN} component={SignIn}/>
        <Route exact path={AppRoute.MY_LIST}
               render={() => (isAuthorized
                 ? <UserPage/>
                 : <SignIn/>)}/>
        <Route
          render={() => (
            <React.Fragment>
              <h1>
                Error: 404. Page not found.
              </h1>
              <Link to={AppRoute.MAIN}>Go to main page</Link>
            </React.Fragment>
          )}
        />
      </Switch>
    </Router>
  );
};

const mapStateToProps = (state) => ({
  films: getFilms(state),
  filteredFilms: getFilteredFilms(state),
  filmsOnScreen: getFilmsOnScreen(state),
  promoFilm: getPromoFilm(state),
  comments: getComments(state),
  isAuthorized: getAuthorizationStatus(state),
  isFilmsLoaded: getFilmsLoadingStatus(state),
  requestStatus: getRequestStatus(state),
  isPromoFilmLoaded: getPromoFilmLoadState(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadComments(filmId) {
    dispatch(DataActionCreator.changeCommentsLoadState(false));
    dispatch(DataOperation.loadComments(filmId));
  },
});

export { App as AppComponent };
export default connect(mapStateToProps, mapDispatchToProps)(App);
