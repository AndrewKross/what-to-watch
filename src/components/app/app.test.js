import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { comments, defaultMockStore, films } from '../../mocks/test-mocks';
import { AppComponent } from './app.jsx';
import { noop } from '../../utils/common';
import { FILMS_COUNT_ON_START, HttpStatus } from '../../const';

const mockStoreConfig = configureStore([]);
const store = mockStoreConfig(defaultMockStore);

describe(`App Component`, () => {
  test(`should render correctly`, () => {
    const tree = renderer.create(
      <Provider store={store}>
        <AppComponent films={films} filmsOnScreen={FILMS_COUNT_ON_START} loadComments={noop}
                      isFilmsLoaded={true} isPromoFilmLoaded={true} promoFilm={films[0]}
                      requestStatus={HttpStatus.SUCCESS} filteredFilms={films}
                      comments={comments} isAuthorized={true}/>
      </Provider>, {
        createNodeMock: () => ({}),
      },
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test(`should render spinner, when loading films`, () => {
    const tree = renderer.create(
      <Provider store={store}>
        <AppComponent films={[]} filmsOnScreen={FILMS_COUNT_ON_START} loadComments={noop}
                      isFilmsLoaded={false} filteredFilms={[]} comments={[]} isAuthorized={true}
                      isPromoFilmLoaded={false} promoFilm={{}} requestStatus={HttpStatus.SUCCESS}/>
      </Provider>, {
        createNodeMock: () => ({}),
      },
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test(`should render error, if loading was failed`, () => {
    const tree = renderer.create(
      <Provider store={store}>
        <AppComponent films={[]} filmsOnScreen={FILMS_COUNT_ON_START} loadComments={noop}
                      isFilmsLoaded={true} filteredFilms={[]} comments={[]} isAuthorized={false}
                      isPromoFilmLoaded={true} promoFilm={{}} requestStatus={HttpStatus.NOT_FOUND}/>
      </Provider>, {
        createNodeMock: () => ({}),
      },
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
