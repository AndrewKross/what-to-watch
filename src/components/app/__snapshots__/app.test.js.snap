// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App should render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <img
            alt="User avatar"
            height="63"
            src="img/1.png"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald"
            height={327}
            src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
            width={218}
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              comedy
            </span>
            <span
              className="movie-card__year"
            >
              2020
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            comedy
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            melodrama
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            horror
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            documentary
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            romance
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            thriller
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/bohemian-rhapsody.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/macbeth.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/aviator.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/we-need-to-talk-about-kevin.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/what-we-do-in-the-shadows.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/revenant.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height={175}
              muted={true}
              poster="img/johnny-english.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width={280}
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
            >
              Johnny English
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="#"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;

exports[`App Component App should render error, if loading was failed 1`] = `
<h1>
  Error: 
  404
</h1>
`;

exports[`App Component App should render spinner, when loading films 1`] = `
<div
  style={
    Object {
      "left": "50%",
      "position": "fixed",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
    }
  }
>
  <div
    aria-busy="true"
    className=""
    style={Object {}}
  >
    <svg
      aria-label="audio-loading"
      fill="#a071bb"
      height={250}
      viewBox="0 0 135 135"
      width={250}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M67.447 58c5.523 0 10-4.477 10-10s-4.477-10-10-10-10 4.477-10 10 4.477 10 10 10zm9.448 9.447c0 5.523 4.477 10 10 10 5.522 0 10-4.477 10-10s-4.478-10-10-10c-5.523 0-10 4.477-10 10zm-9.448 9.448c-5.523 0-10 4.477-10 10 0 5.522 4.477 10 10 10s10-4.478 10-10c0-5.523-4.477-10-10-10zM58 67.447c0-5.523-4.477-10-10-10s-10 4.477-10 10 4.477 10 10 10 10-4.477 10-10z"
      >
        <animateTransform
          attributeName="transform"
          dur="2.5s"
          from="0 67 67"
          repeatCount="indefinite"
          to="-360 67 67"
          type="rotate"
        />
      </path>
      <path
        d="M28.19 40.31c6.627 0 12-5.374 12-12 0-6.628-5.373-12-12-12-6.628 0-12 5.372-12 12 0 6.626 5.372 12 12 12zm30.72-19.825c4.686 4.687 12.284 4.687 16.97 0 4.686-4.686 4.686-12.284 0-16.97-4.686-4.687-12.284-4.687-16.97 0-4.687 4.686-4.687 12.284 0 16.97zm35.74 7.705c0 6.627 5.37 12 12 12 6.626 0 12-5.373 12-12 0-6.628-5.374-12-12-12-6.63 0-12 5.372-12 12zm19.822 30.72c-4.686 4.686-4.686 12.284 0 16.97 4.687 4.686 12.285 4.686 16.97 0 4.687-4.686 4.687-12.284 0-16.97-4.685-4.687-12.283-4.687-16.97 0zm-7.704 35.74c-6.627 0-12 5.37-12 12 0 6.626 5.373 12 12 12s12-5.374 12-12c0-6.63-5.373-12-12-12zm-30.72 19.822c-4.686-4.686-12.284-4.686-16.97 0-4.686 4.687-4.686 12.285 0 16.97 4.686 4.687 12.284 4.687 16.97 0 4.687-4.685 4.687-12.283 0-16.97zm-35.74-7.704c0-6.627-5.372-12-12-12-6.626 0-12 5.373-12 12s5.374 12 12 12c6.628 0 12-5.373 12-12zm-19.823-30.72c4.687-4.686 4.687-12.284 0-16.97-4.686-4.686-12.284-4.686-16.97 0-4.687 4.686-4.687 12.284 0 16.97 4.686 4.687 12.284 4.687 16.97 0z"
      >
        <animateTransform
          attributeName="transform"
          dur="8s"
          from="0 67 67"
          repeatCount="indefinite"
          to="360 67 67"
          type="rotate"
        />
      </path>
    </svg>
  </div>
</div>
`;
